src/services/review/ReviewRequestService.ts

```diff
@@ -481,47 +487,57 @@ export default class ReviewRequestService {
     sessionData: UserWithSiteSessionData,
     site: Site
   ): Promise<void> => {
-    const { isomerUserId: userId } = sessionData
-
-    const requestsViewed = await this.reviewRequestView.findAll({
-      where: {
-        siteId: site.id,
-        userId,
-      },
-    })
+    try {
+      const { isomerUserId: userId } = sessionData
+      await this.sequelize.transaction(async (transaction) => {
+        const requestsViewed = await this.reviewRequestView.findAll({
+          where: {
+            siteId: site.id,
+            userId,
+          },
+          transaction,
+        })
 
-    const allActiveRequests = await this.repository.findAll({
-      where: {
-        siteId: site.id,
-        // NOTE: Closed and merged review requests would not have an
-        // entry in the review request views table
-        reviewStatus: ["OPEN", "APPROVED"],
-      },
-    })
+        const allActiveRequests = await this.repository.findAll({
+          where: {
+            siteId: site.id,
+            reviewStatus: ["OPEN", "APPROVED"],
+          },
+          transaction,
+        })
 
-    const requestIdsViewed = requestsViewed.map(
-      (request) => request.reviewRequestId
-    )
-    const allActiveRequestIds = allActiveRequests.map((request) => request.id)
-    const requestIdsToMarkAsViewed = _.difference(
-      allActiveRequestIds,
-      requestIdsViewed
-    )
+        const requestIdsViewed = requestsViewed.map(
+          (request) => request.reviewRequestId
+        )
+        const allActiveRequestIds = allActiveRequests.map(
+          (request) => request.id
+        )
+        const requestIdsToMarkAsViewed = _.difference(
+          allActiveRequestIds,
+          requestIdsViewed
+        )
 
-    await Promise.all(
-      // Using map here to allow creations to be done concurrently
-      // But we do not actually need the result of the view creation
-      requestIdsToMarkAsViewed.map(async (requestId) =>
-        this.reviewRequestView.create({
-          reviewRequestId: requestId,
-          siteId: site.id,
-          userId,
-          // This field represents the user opening the review request
-          // itself, which the user has not done so yet at this stage.
-          lastViewedAt: null,
-        })
-      )
-    )
+        await Promise.all(
```

Comment:
just realised that this is cos of repeated attempts to create - in this case, the `transaction` might not help at all actually because we'd still try to create the repeated entry. 

maybe we should just use `_.unique(requestIdsToMarkAsViewed)` to prevent dups and accept hte race condition of having some views being outdated (actually seen but presented as not), as it's relatively minor.

Reply:
> in this case, the transaction might not help at all actually because we'd still try to create the repeated entry.

hmm i am confused here, why would it try to create a repeated entry ya? there is still a race condition here, but in this case we are "safely" failing here right due to a rollback? so rather than making our server return 500, we are increasing the stability of our be here no? 

> maybe we should just use _.unique(requestIdsToMarkAsViewed) to prevent dups and accept hte race condition of having some views being outdated (actually seen but presented as not), as it's relatively minor.

hmm wait ah this is due to duplicate api calls leh, how would this help here ah?

Reply:
i think the error stems from an attempt to create when there's an existing entry + a unique constraint. this is **not** due to concurrency issues and you can replicate this just by having the `requestIds` to be identical and do the creation sequentially.

because the error is due to duplicates in `requestIds`, the fix is to either remove the duplicates or allow the update but have it do nth.

Reply:
wait we do have an initial call here right, in line 493? 
```
 const requestsViewed = await this.reviewRequestView.findAll({
          where: {
            siteId: site.id,
            userId,
          },
          transaction,
        })
```
we get the list of requestsViews, diff it using the current open review requests, and iff there are active requests that **_have not been_** viewed yet, create a entry in the `reviewRequestViews`?

Reply:
if you've tried that this solution works on local, then please merge it. i might have a misconception, so it shouldn't block the PR